src/mscorlib/src/System/TimeZoneInfo.cs

```diff
@@ -1076,6 +1099,7 @@ private AdjustmentRule GetAdjustmentRuleForTime(DateTime dateTime, bool dateTime
                 int compareResult = CompareAdjustmentRuleToDateTime(rule, previousRule, dateTime, date, dateTimeisUtc);
                 if (compareResult == 0)
                 {
+                    ruleIndex = median;
```

Comment:
I thought of it, but that would involve changing serialization of these rules, which I wasn't prepared to do with this change.  Or we would have to set the index in all the constructors (both normal and during serialization).  I just felt this was a more targeted change.